Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/OSIRISDiffractionReduction.py

```diff
@@ -175,23 +218,58 @@ def get_intersection_of_ranges(range_list):
     the same point.  Also, all ranges should obey range[0] <= range[1].
     """
     # Find all combinations of ranges, and see where they intersect.
-    rangeCombos = list(itertools.combinations(range_list, 2))
-    intersections = []
-    for rangePair in rangeCombos:
-        intersection = find_intersection_of_ranges(rangePair[0], rangePair[1])
-        if intersection is not None:
-            intersections.append(intersection)
+    range_combos = itertools.combinations(range_list, 2)
+
+    # Retrieve all intersections
+    intersections = map(lambda range_pair: find_intersection_of_ranges(range_pair[0], range_pair[1]),
+                        range_combos)
+
+    # Filter out None type intersections
+    intersections = filter(lambda intersection: intersection is not None, intersections)
 
     # Return the sorted intersections.
-    intersections.sort()
+    intersections = sorted(intersections)
     return intersections
 
 
 def is_in_ranges(range_list, val):
-    for myrange in range_list:
-        if myrange[0] < val < myrange[1]:
-            return True
-    return False
+    return any(arange[0] < val < arange[1] for arange in range_list)
+
+
+def list_to_range(array_to_convert):
+    """
+    Converts a specified array to a range representation, based on the
+    following specification -
+
+    Array with one element          : Array is returned immediately
+    Array with two or more elements : A range from the first to second element
+                                      (inclusive) is returned.
+
+    :param array_to_convert:  The array to convert to a range.
+    :return:                The generated range.
+    """
+
+    if len(array_to_convert) == 1:
+        return array_to_convert
+    else:
+        return range(array_to_convert[0], array_to_convert[1] + 1)
+
+
+def rebin_to_smallest(workspaces):
+    """
+    Rebins the specified list to the workspace with the smallest
+    x-range in the list.
+
+    :param workspaces: The list of workspaces to rebin to the smallest.
+    :return:           The rebinned list of workspaces.
+    """
+    smallest_ws = min(workspaces, key=lambda ws: mtd[ws].blocksize())
+
+    return [RebinToWorkspace(
```

Comment:
Could you change this to a for loop? Also maybe change Rebin to a child algorith.